api.lua

```diff
@@ -0,0 +1,26 @@
+
+function goodtorch.detect_gamemode()
+    if minetest.registered_node["default:dirt"] then
+        return "MTG" -- Minetest Game
+    elseif minetest.registered_node["mcl_core:dirt"] then
+        return "MCL" -- Some version of MineClone (2, 5, Mineclonia, don't know, or care)
+    else
+        return "???" -- Unknown game (we don't want to see this, this is bad, could be on a different game we are not supporting/yet)
+    end
+end
```

Comment:
This attempts to guess what the gamemode/game is. (Since there sadly isn't a clean/clear function to call to know the exact gamemode/game)

If we are under MineClone this is new turf, but most of the code should be fine (perhaps a graphical glitch but meh, graphics can be dealt with later down the road).

If Minetest Game (default) is present the mod runs just like it always would.